src/Functions/array/arrayElement.cpp

```diff
@@ -258,7 +262,7 @@ struct ArrayElementNumImpl
 
 struct ArrayElementStringImpl
 {
-    template <bool negative>
+    template <bool negative, bool used_builder>
```

Comment:
My only concern is this extra template parameter. Why is it better then using the passed-in `builder` object? As far as I see we can avoid some `if (builder)` checks in runtime, but I am not sure if the extra complexity worth the gains.

Reply:
@lgbo-ustc what do you think about this?

Reply:
Just want to reduce the branch jumps in a loop.